GLMakie/src/GLAbstraction/GLTypes.jl

```diff
@@ -303,12 +314,46 @@ mutable struct RenderObject{Pre}
         # But with this implementation, the fxaa flag can't be changed,
         # and since this is a UUID, it shouldn't matter
         id = pack_bool(RENDER_OBJECT_ID_COUNTER[], fxaa)
-        new(
+        robj = new(
             context,
             uniforms, observables, vertexarray,
             prerenderfunctions, postrenderfunctions,
-            id
+            id, true
         )
+
+        if track_updates
+            # gather update requests for polling in renderloop
+            for uniform in values(uniforms)
+                if uniform isa Observable
+                    on(uniform) do _
+                        robj.requires_update = true
+                    end
+                elseif uniform isa GPUArray
+                    on(uniform.requires_update) do _
+                        robj.requires_update = true
+                    end
+                end
+            end
+            on(_ -> robj.requires_update = true, vertexarray.requires_update)
+        else
+            # remove tracking from GPUArrays
```

Comment:
not sure why that's needed, could you elaborate the comment?

Reply:
Hm, it feels cleaner to me, to just let the GPUArrays track their updates, which should be very cheap, but then not connect them further if we don't track updates...

Reply:
I think I actually removed all the tracking in one go so I'm not sure how much the GPUArrays contribute. The test case I used also didn't trigger updates there, but in simple observables since that should have a higher ratio of tracking cost / total update cost and I wanted to know how bad it can get. 

I was planning to test having GPUArray updates wait for frames in the future, which would probably require something like `requires_update` too. So maybe this will end up as always on/tracking later anyway...

Reply:
```julia
using BenchmarkTools, GLMakie

scene = Scene()
for _ in 1:100
    heatmap!(scene, rand(16, 16))
end
display(GLMakie.Screen(render_on_demand = false), scene)

function update!(scene)
    for p in scene.plots
        notify(p[3])
    end
end

@benchmark update!($scene)
```

### With cleanup

```
BenchmarkTools.Trial: 9326 samples with 1 evaluation.
 Range (min … max):  381.502 μs …  32.987 ms  ┊ GC (min … max): 0.00% … 98.11%
 Time  (median):     502.491 μs               ┊ GC (median):    0.00%
 Time  (mean ± σ):   532.654 μs ± 812.194 μs  ┊ GC (mean ± σ):  3.84% ±  2.49%

      ▁           ▃▄▇█▄▃▁▁▁▂▁▁                                   
  ▁▁▅▄█▅▄▄▄▃▄▄▄▄▄▆█████████████▇▇▇▆▇▆▆▅▅▅▅▅▄▄▄▃▃▃▃▃▃▂▂▂▂▂▂▁▂▁▁▁ ▄
  382 μs           Histogram: frequency by time          694 μs <

 Memory estimate: 121.88 KiB, allocs estimate: 900.
```

### Without cleanup (the else branch commented out)

```
BenchmarkTools.Trial: 4238 samples with 1 evaluation.
 Range (min … max):  803.314 μs …  34.882 ms  ┊ GC (min … max): 0.00% … 96.12%
 Time  (median):       1.153 ms               ┊ GC (median):    0.00%
 Time  (mean ± σ):     1.174 ms ± 745.947 μs  ┊ GC (mean ± σ):  1.33% ±  2.08%

          ▂▃▅▅▅█▃▃▃▄▃▃▃▃▄▄▄▃▃▄▃▃▇▃▅▇▄▃▄▄▃▃▁▂                     
  ▁▄▄▆▄▅▅▃███████████████████████████████████▆▇▆▅▅▄▄▃▄▄▃▂▂▃▃▂▂▂ ▅
  803 μs           Histogram: frequency by time          1.6 ms <

 Memory estimate: 128.12 KiB, allocs estimate: 1300.
```

### Without cleanup & single observer function

```
BenchmarkTools.Trial: 4747 samples with 1 evaluation.
 Range (min … max):  757.975 μs …  31.849 ms  ┊ GC (min … max): 0.00% … 96.00%
 Time  (median):       1.006 ms               ┊ GC (median):    0.00%
 Time  (mean ± σ):     1.048 ms ± 783.416 μs  ┊ GC (mean ± σ):  1.84% ±  2.42%

          ▃▄▇▅▅█▅▅▅▂▃ ▃▁▁▁▁ ▁▁▂▁▂▁▁  ▁                           
  ▅▃▆▅▄▆▅▆████████████████████████████████▆▆▆▄▄▅▆▄▄▄▃▃▃▃▂▂▂▂▁▂▂ ▅
  758 μs           Histogram: frequency by time         1.43 ms <

 Memory estimate: 128.12 KiB, allocs estimate: 1300.
```

That's much more than I thought it would be...